frontend/src/scenes/experiments/Experiment.tsx

```diff
@@ -918,55 +937,31 @@ export function Experiment(): JSX.Element {
-                                props={{
-                                    dashboardItemId: experimentResults.fakeInsightId,
-                                    cachedInsight: {
-                                        short_id: experimentResults.fakeInsightId,
-                                        filters: {
-                                            ...experimentResults.filters,
-                                            insight: experimentInsightType,
-                                            ...(experimentInsightType === InsightType.FUNNELS && {
-                                                layout: FunnelLayout.vertical,
-                                                funnel_viz_type: FunnelVizType.Steps,
-                                            }),
-                                            ...(experimentInsightType === InsightType.TRENDS && {
-                                                display: ChartDisplayType.ActionsLineGraphCumulative,
-                                            }),
-                                        },
-                                        result: experimentResults.insight,
-                                        disable_baseline: true,
-                                    },
-                                    doNotLoad: true,
-                                }}
-                            >
-                                <div className="mt-4">
-                                    <Query
-                                        query={{
-                                            kind: NodeKind.InsightVizNode,
-                                            source: filtersToQueryNode(
-                                                transformResultFilters(experimentResults.filters)
-                                            ),
-                                            showTable: true,
-                                            showLegendButton: false,
-                                        }}
-                                        context={{
-                                            insightProps: {
-                                                dashboardItemId: experimentResults.fakeInsightId as InsightShortId,
-                                                cachedInsight: {
-                                                    short_id: experimentResults.fakeInsightId as InsightShortId,
-                                                    filters: transformResultFilters(experimentResults.filters),
-                                                    result: experimentResults.insight,
-                                                    disable_baseline: true,
-                                                },
-                                                doNotLoad: true,
+                            <div className="mt-4">
+                                <Query
+                                    query={{
```

Comment:
@thmsobrmlr came across an interesting bug related to Query node which I haven't been able to solve yet:

1. This <Query /> component generates and <InsightViz> component: https://github.com/PostHog/posthog/blob/master/frontend/src/queries/nodes/InsightViz/InsightViz.tsx#L31 with the right dataNodelogic props

2. https://github.com/PostHog/posthog/blob/master/frontend/src/queries/nodes/InsightViz/InsightViz.tsx#L57 -> in this BindLogic, the query for lastRefresh is always empty, unless I refresh the page: https://github.com/PostHog/posthog/blob/master/frontend/src/queries/nodes/DataNode/dataNodeLogic.ts#L366

3. This leads to delayed computation of the last calculated results.

(in this screen cast, before I switch to a different screen, you can see it immediately loading, that's because the refresh loop ran to update these numbers).

![2023-07-06 13 01 10](https://github.com/PostHog/posthog/assets/7115141/89871cdf-926e-43f4-88c1-a8ba5bbddaf4)


Maybe I am doing something wrong here, any ideas?

Reply:
oh interesting, it's a race condition, if I mount it earlier, it's all fine 🤔

Reply:
Ahhh damn it, it's because of this, which is overriding the query to be empty: https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/insights/insightVizDataLogic.ts#L67

Reply:
Not sure why this happens like it does, when BindLogic should've come first and mounted the right logic - but checking the debugger stack tells us that props do change, the insightNodeLogic overrides props for the dataNodeLogic with the same key, thus resulting in last refresh having an empty query node, returning null.

Had a chat with @thmsobrmlr, we've fixed it by removing the dataNode check for last refresh - doesn't seem like it's used.